File: latest_ran_output/qwen3_coder_next/db_004_naive_backend.js
const express = require('express');
const router = express.Router();

// Mock database - in production, this would be a real database connection
const productsDB = [
    { id: 1, name: 'Laptop', category: 'Electronics', price: 999.99 },
    { id: 2, name: 'Smartphone', category: 'Electronics', price: 699.99 },
    { id: 3, name: 'Headphones', category: 'Electronics', price: 149.99 },
    { id: 4, name: 'T-Shirt', category: 'Clothing', price: 29.99 },
    { id: 5, name: 'Jeans', category: 'Clothing', price: 59.99 },
    { id: 6, name: 'Novel', category: 'Books', price: 14.99 },
    { id: 7, name: 'Cookbook', category: 'Books', price: 24.99 },
    { id: 8, name: 'Garden Tools Set', category: 'Home & Garden', price: 89.99 }
];

// GET /api/products - Filter products based on query parameters
router.get('/', (req, res) => {
    try {
        const { category, minPrice = 0, maxPrice = Infinity } = req.query;

        // Convert price values to numbers
        const minPriceNum = parseFloat(minPrice);
        const maxPriceNum = parseFloat(maxPrice);

        // Filter products based on criteria
        let filteredProducts = productsDB.filter(product => {
            // Category filter (case-insensitive)
            if (category && product.category.toLowerCase() !== category.toLowerCase()) {
                return false;
            }

            // Price range filter
            if (product.price < minPriceNum || product.price > maxPriceNum) {
                return false;
            }

            return true;
        });

        res.json(filteredProducts);
    } catch (error) {
        console.error('Error filtering products:', error);
        res.status(500).json({ error: 'Internal server error' });
    }
});

module.exports = router;

File: latest_ran_output/qwen3_coder_next/db_004_naive_backend.py
from flask import Flask, request, jsonify
from sqlalchemy import create_engine, Column, Integer, String, Float
from sqlalchemy.ext.declarative import declarative_base
from sqlalchemy.orm import sessionmaker

app = Flask(__name__)
Base = declarative_base()

# Define the Product model (adjust connection string as needed)
class Product(Base):
    __tablename__ = 'products'
    
    id = Column(Integer, primary_key=True)
    name = Column(String(100))
    category = Column(String(50))
    price = Column(Float)

# Example database setup (replace with your actual DB connection)
engine = create_engine('sqlite:///products.db')  # For demo purposes
Session = sessionmaker(bind=engine)
session = Session()

@app.route('/products', methods=['GET'])
def get_products():
    try:
        # Get query parameters
        category = request.args.get('category')
        min_price = request.args.get('min_price')
        max_price = request.args.get('max_price')
        
        # Build the base query
        query = session.query(Product)
        
        # Apply filters if provided
        if category:
            query = query.filter(Product.category == category)
            
        if min_price is not None:
            try:
                min_price_val = float(min_price)
                query = query.filter(Product.price >= min_price_val)
            except ValueError:
                return jsonify({"error": "Invalid min_price value"}), 400
                
        if max_price is not None:
            try:
                max_price_val = float(max_price)
                query = query.filter(Product.price <= max_price_val)
            except ValueError:
                return jsonify({"error": "Invalid max_price value"}), 400
        
        # Execute the query
        products = query.all()
        
        # Convert to JSON-serializable format
        result = []
        for product in products:
            result.append({
                "id": product.id,
                "name": product.name,
                "category": product.category,
                "price": product.price
            })
            
        return jsonify(result)
    
    except Exception as e:
        return jsonify({"error": str(e)}), 500

# For demo purposes only - create tables and add sample data if needed
if __name__ == '__main__':
    Base.metadata.create_all(engine)
    # Add some sample products if table is empty
    if session.query(Product).count() == 0:
        sample_products = [
            Product(name="Laptop", category="Electronics", price=999.99),
            Product(name="Headphones", category="Electronics", price=149.50),
            Product(name="Coffee Mug", category="Kitchen", price=12.99),
            Product(name="Notebook", category="Stationery", price=5.99)
        ]
        session.add_all(sample_products)
        session.commit()
    
    app.run(debug=True)